<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ClientChatWindow</class>
 <widget class="QMainWindow" name="ChatWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Chat Window</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
    background-color: #f5f5f5;
}
QWidget#centralwidget {
    background-color: #f5f5f5;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_main">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="clientInfoSection" native="true">
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>280</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true">QWidget#clientInfoSection {
    background-color: #ffffff;
    border-left: 1px solid #e0e0e0;
}</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>15</number>
       </property>
       <property name="leftMargin">
        <number>20</number>
       </property>
       <property name="topMargin">
        <number>40</number>
       </property>
       <property name="rightMargin">
        <number>20</number>
       </property>
       <property name="bottomMargin">
        <number>20</number>
       </property>
       <item alignment="Qt::AlignmentFlag::AlignHCenter">
        <widget class="QLabel" name="clientLogoLabel">
         <property name="minimumSize">
          <size>
           <width>120</width>
           <height>120</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>120</width>
           <height>120</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QLabel {
    background-color: #f0f0f0;
    border: 3px solid #d0d0d0;
    border-radius: 60px;
}</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="clientOrgLabel">
         <property name="styleSheet">
          <string notr="true">font-size: 14pt; font-weight: bold; color: #333;</string>
         </property>
         <property name="text">
          <string>Organization Logo</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="styleSheet">
          <string notr="true">QFrame {
    background-color: #d0d0d0;
    min-height: 1px;
    max-height: 1px;
}</string>
         </property>
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="clientConnectionLabel">
         <property name="styleSheet">
          <string notr="true">font-size: 11pt; color: #555; padding: 5px;</string>
         </property>
         <property name="text">
          <string>Connection Status: Connected</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="reconnectBtn">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QPushButton {
    background-color: #075e54;
    color: white;
    border: none;
    border-radius: 8px;
    font-size: 11pt;
    padding: 8px;
}
QPushButton:hover {
    background-color: #0a7c6f;
}
QPushButton:pressed {
    background-color: #055346;
}</string>
         </property>
         <property name="text">
          <string>Reset and Reconnect</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="chatSection" native="true">
      <property name="styleSheet">
       <string notr="true">QWidget#chatSection {
    background-color: #ffffff;
}</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="chatHeader" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>70</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>70</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QWidget#chatHeader {
    background-color: #075e54;
    border-bottom: 1px solid #055346;
}</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="chatTitleLabel">
            <property name="styleSheet">
             <string notr="true">font-size: 14pt; font-weight: bold; padding: 0px; color: white;</string>
            </property>
            <property name="text">
             <string>Chat with Server</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="chatSubtitleLabel">
            <property name="styleSheet">
             <string notr="true">font-size: 10pt; padding: 2px; color: #ffffff;</string>
            </property>
            <property name="text">
             <string>Connected to Server</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="chatHistoryWdgt">
         <property name="layoutDirection">
          <enum>Qt::LayoutDirection::LeftToRight</enum>
         </property>
         <property name="styleSheet">
          <string notr="true">QListWidget {
    background-color: #e5ddd5;
    border: none;
    font-size: 13pt;
    padding: 10px;
    color: #000000;
}

/* * --- FIX ---
 * Make the item background transparent so the custom widget's style shows through.
 * Remove border, padding, margin from the item itself.
*/
QListWidget::item {
    background-color: transparent;
    border: none;
    padding: 0px;
    margin: 0px;
}
QListWidget::item:selected,
QListWidget::item:selected:active,
QListWidget::item:selected:!active,
QListWidget::item:selected:hover {
    background-color: transparent;
    border: none;
    outline: none;
}
/* --- END FIX --- */
</string>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
         </property>
         <property name="spacing">
          <number>5</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="inputSection" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>80</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>80</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QWidget#inputSection {
    background-color: #f0f0f0;
    border-top: 1px solid #d0d0d0;
}</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="spacing">
           <number>10</number>
          </property>
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <property name="bottomMargin">
           <number>10</number>
          </property>
          <item>
           <widget class="QTextEdit" name="typeMessageTxt">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>60</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>60</height>
             </size>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LayoutDirection::LeftToRight</enum>
            </property>
            <property name="styleSheet">
             <string notr="true">QTextEdit {
    background-color: white;
    color: #000000;
    border: 1px solid #d0d0d0;
    border-radius: 20px;
    padding: 10px 15px;
    font-size: 13pt;
}
QTextEdit:focus {
    border: 2px solid #25d366;
}</string>
            </property>
            <property name="placeholderText">
             <string>Type your message...</string>
            </property>
           </widget>
          </item>
      <item>
       <widget class="QPushButton" name="sendFileBtn">
            <property name="minimumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton {
    background-color: #075e54;
    color: white;
    border: none;
    border-radius: 8px;
    font-size: 14pt;
    font-weight: bold;
}
QPushButton:hover {
    background-color: #0a7c6f;
}
QPushButton:pressed {
    background-color: #055346;
}</string>
            </property>
            <property name="text">
             <string>+</string>
            </property>
           </widget>
          </item>
      <item>
       <widget class="QPushButton" name="recordVoiceBtn">
        <property name="minimumSize">
         <size>
          <width>80</width>
          <height>60</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>80</width>
          <height>60</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Record a voice message</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton {
    background-color: #128c7e;
    color: white;
    border: none;
    border-radius: 8px;
    font-size: 13pt;
    font-weight: bold;
}
QPushButton:hover {
    background-color: #0f7569;
}
QPushButton:pressed {
    background-color: #0c5c52;
}</string>
        </property>
        <property name="text">
         <string>Rec</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="sendMessageBtn">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>60</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>80</width>
              <height>60</height>
             </size>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton {
    background-color: #25d366;
    color: white;
    border: none;
    border-radius: 8px;
    font-size: 14pt;
    font-weight: bold;
}
QPushButton:hover {
    background-color: #20bd5a;
}
QPushButton:pressed {
    background-color: #1da851;
}</string>
            </property>
            <property name="text">
             <string>Send</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
